pkg/prom/ha/server.go

```diff
@@ -159,6 +163,20 @@ func New(reg prometheus.Registerer, cfg Config, globalConfig *config.GlobalConfi
 	return s, nil
 }
 
+// newRing creates a new Cortex Ring that ignores unhealthy nodes.
+func newRing(cfg ring.Config, name, key string, reg prometheus.Registerer) (*ring.Ring, error) {
+	codec := ring.GetCodec()
+	store, err := kv.NewClient(
+		cfg.KVStore,
+		codec,
+		kv.RegistererWithKVName(reg, name+"-ring"),
+	)
+	if err != nil {
+		return nil, err
+	}
+	return ring.NewWithStoreClientAndStrategy(cfg, name, key, store, ring.NewIgnoreUnhealthyInstancesReplicationStrategy())
```

Comment:
This is just the implementation of `ring.New` but with the replication strategy swapped out. Convenient that Cortex had exactly the strategy I wanted!